### PR TITLE
deps,config,factory: update CDI deps, use updated CDI interface.

### PR DESCRIPTION
**Pull request by klihub**
#### What type of PR is this?

/kind other

#### What this PR does / why we need it:

Update CDI dependency to the latest tagged version (v0.7.2). Update our CDI handling code to use the preferred revisited top-level CDI interface.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by kannon92** [comment]
Related to https://github.com/cncf-tags/container-device-interface/issues/199.

cc @elezar 
for cdi changes.

LGTM to me.

**Comment by kannon92** [comment]
@cri-o/cri-o-maintainers PTAL.

According to the issue I linked, we may also want to consider this for 1.30.

**Comment by klihub** [comment]
> @cri-o/cri-o-maintainers PTAL.
> 
> According to the issue I linked, we may also want to consider this for 1.30.

@kannon92: Yes, I'd like to get this into release-1.30 as well. If that is the way to go, I can file a similar PR against the 1.30 release branch once this is deemed good to go in.

**Comment by klihub** [comment]
`ci-cgroup-v2-integration` seems to pretty consistently fail now with `not ok 174 run container in pod with crun-wasm enabled` for `test/image.bats`. The errors seems to always be `unknown version specified\n` in `oci/runtime_oci.go:306`, but I can neither reproduce it locally nor figure out (at least with a few simple `git grep`s in crun, WasmEdge, etc.) where that error message is coming from... any ideas ?

**Comment by klihub** [comment]
> `ci-cgroup-v2-integration` seems to pretty consistently fail now with `not ok 174 run container in pod with crun-wasm enabled` for `test/image.bats`. The errors seems to always be `unknown version specified\n` in `oci/runtime_oci.go:306`, but I can neither reproduce it locally nor figure out (at least with a few simple `git grep`s in crun, WasmEdge, etc.) where that error message is coming from... any ideas ?

Ditto for `ci-fedora-integration`...

**Comment by kannon92** [comment]
@klihub we are actually working on this one. We are seeing in the podman 5.0 version bump. It comes from an old version of crun.

**Comment by kannon92** [comment]
could you try a rebase? We pushed an update to fix the CI so want to see if this works.

**Comment by klihub** [comment]
@kannon92 I see the same `run container in pod with crun-wasm enabled` failures for `ci-fedora-integration` and `ci-cgroup-v2-integration`...

**Comment by klihub** [comment]
> could you try a rebase? We pushed an update to fix the CI so want to see if this works.

@kannon92 Hmm... rebased again on latest main/HEAD and still see the same failures. Which commit(s) should provide the CI fix ? Do I have them in my tree ?

**Comment by kannon92** [comment]
Sorry, my hopes were dashed. We will continue to look into this fix.

**Comment by klihub** [comment]
> Sorry, my hopes were dashed. We will continue to look into this fix.

Ah, ok. Ping me if there is something new to test, and I'll rebase again.

**Comment by kannon92** [comment]
> > Sorry, my hopes were dashed. We will continue to look into this fix.
> 
> Ah, ok. Ping me if there is something new to test, and I'll rebase again.

We pushed https://github.com/cri-o/cri-o/pull/7988 to skip this test for now. Please rebase when you can!

**Comment by kannon92** [comment]
:( these crun tests are the bane of my existence right now. Still working on them.

**Comment by kannon92** [comment]
Thanks to @sohankunkerkar, I think this is good to go now.

**Comment by klihub** [comment]
> > Thanks to @sohankunkerkar, I think this is good to go now.
> 
> Yes, but it as with runtime-spec forced to c0e90434df2a. Let's see will it start failing (those few openshift CI runts) if I revert it back to 1.2.0...

Yup. So I re-pushed a bit cleaner version of the runtime-spec version downgrade commit. We either need to go with this or downgrade runtime-spec in CDI, tag a new 0.7.2 version out of that, and use that here.

**Comment by klihub** [comment]
@kannon92 @sohankunkerkar @saschagrunert This PR now forces runtime-spec to the previously used untagged version as a temporary workaround (for CDI forcing 1.2.0 on us, which breaks a few test runs in openshift) until we figure out what prevents us from updating to runtime-spec 1.2.0. If this is not acceptable please chime in...

**Comment by klihub** [comment]
If #7998 passes all tests, I'll rebase this, remove the forced downgrade kludge and push again to see if it helps...

**Comment by saschagrunert** [comment]
> This PR now forces runtime-spec to the previously used untagged version as a temporary workaround

Hm, I personally prefer fixing than working around. This seems to be a critical CI issue.

**Comment by klihub** [comment]
> > This PR now forces runtime-spec to the previously used untagged version as a temporary workaround
> 
> Hm, I personally prefer fixing than working around. This seems to be a critical CI issue.

Okay, so I remove the workaround kludge, then rebase once/until we've managed to sort out the CI failures.

File: go.mod [hunk 1]
```diff
@@ -83,7 +83,7 @@ require (
 	sigs.k8s.io/release-sdk v0.11.0
 	sigs.k8s.io/release-utils v0.7.7
 	sigs.k8s.io/yaml v1.4.0
-	tags.cncf.io/container-device-interface v0.6.2
+	tags.cncf.io/container-device-interface v0.7.2-0.20240416155956-d7f87365dfce
```
Comment (kannon92) [inline go.mod:86]: You probably want to also downgrade runtime spec.

Comment (kannon92) [reply go.mod:86]: sorry you aren't touching that here.

Comment (kannon92) [reply go.mod:86]: Do we want to wait until there is an official version for cdi?

**Comment by elezar** [comment]
I have pushed a new CDI release: https://github.com/cncf-tags/container-device-interface/releases/tag/v0.7.2 feel free to update it here.

File: go.mod (hunk 1 above)
Comment (kannon92) [inline go.mod:86]:
```suggestion
	tags.cncf.io/container-device-interface v0.7.2
```

Comment (klihub) [reply go.mod]: I just wanted to see whether this goes all green without waiting for a new tag. But now we have the new tag, so I update our deps to refer to v0.7.2

**Review by saschagrunert — APPROVED** [review]

**Pull request merged**